### PR TITLE
Refine PADDLE_ENFORCE

### PR DESCRIPTION
**Pull request by GaoWei8**
### PR types
Others

### PR changes
Others

### Describe
修复paddle/fluid/platform路径下，PADDLE_ENFORCE的不合规报错。

### CODE REVIEW

File: paddle/fluid/platform/cpu_helper.cc [hunk 1]
```diff
@@ -43,7 +43,11 @@ void SetNumThreads(int num_threads) {
   platform::dynload::MKL_Set_Num_Threads(real_num_threads);
   omp_set_num_threads(real_num_threads);
 #else
-  PADDLE_ENFORCE(false, "To be implemented.");
+  PADDLE_ENFORCE_EQ(
```
Comment (chenwhql) [inline paddle/fluid/platform/cpu_helper.cc]: maybe here can use `PADDLE_THROW` directly?

File: paddle/fluid/platform/cudnn_helper.h [hunk 2]
```diff
@@ -103,7 +103,7 @@ inline cudnnPoolingMode_t GetPoolingMode(const PoolingMode& mode) {
     case PoolingMode::kMaximum:
       return CUDNN_POOLING_MAX;
     default:
-      PADDLE_THROW("Unexpected pooling mode.");
+      PADDLE_THROW(platform::errors::Unimplemented("Unexpected pooling mode."));
```
Comment (chenwhql) [inline paddle/fluid/platform/cudnn_helper.h]: `Unexpeccted CUDNN pooling mode.` maybe better

File: paddle/fluid/platform/cudnn_helper.h [hunk 3]
```diff
@@ -119,7 +119,7 @@ inline cudnnPoolingMode_t GetPoolingMode(const PoolingMode& mode) {
...
```
Comment (chenwhql) [inline paddle/fluid/platform/cudnn_helper.h]: same above

File: paddle/fluid/platform/cudnn_helper.h [hunk 4]
```diff
@@ -208,7 +209,8 @@ inline cudnnTensorFormat_t GetCudnnTensorFormat(
     case DataLayout::kNDHWC:
       return CUDNN_TENSOR_NHWC;  // add, liyamei
     default:
-      PADDLE_THROW("Unknown cudnn equivalent for order");
+      PADDLE_THROW(platform::errors::Unimplemented(
+          "Unknown cudnn equivalent for order."));
```
Comment (chenwhql) [inline paddle/fluid/platform/cudnn_helper.h]: do not understand this message

File: paddle/fluid/platform/cudnn_helper.h [hunk 5]
```diff
@@ -456,8 +477,9 @@ class ScopedActivationDescriptor {
         mode = CUDNN_ACTIVATION_TANH;
         break;
       default:
-        PADDLE_THROW("unrecognized activation mode: %d .",
-                     static_cast<int>(activation_mode));
+        PADDLE_THROW(
+            platform::errors::Unimplemented("unrecognized activation mode: %d.",
```
Comment (chenwhql) [inline paddle/fluid/platform/cudnn_helper.h]: `unrecognized activation mode` -> `Unrecognized CuDNN activation mode`?

File: paddle/fluid/platform/device_context.cc [hunk 6]
```diff
@@ -84,7 +83,11 @@ inline void EmplaceDeviceContext(
 
 DeviceContextPool::DeviceContextPool(
     const std::vector<platform::Place>& places) {
-  PADDLE_ENFORCE_GT(places.size(), 0);
+  PADDLE_ENFORCE_GT(
+      places.size(), 0,
+      platform::errors::InvalidArgument(
+          "The number of platform places should >0. But received %d.",
```
Comment (chenwhql) [inline paddle/fluid/platform/device_context.cc]: `>` use english word is better

File: paddle/fluid/platform/dynload/dynamic_loader.cc [hunk 7]
```diff
@@ -319,9 +319,9 @@ void* GetMKLMLDsoHandle() {
 
 void* GetOpDsoHandle(const std::string& dso_name) {
 #if defined(__APPLE__) || defined(__OSX__)
-  PADDLE_THROW("Do not support Apple.");
+  PADDLE_THROW(platform::errors::Unimplemented("Do not support Apple."));
```
Comment (chenwhql) [inline paddle/fluid/platform/dynload/dynamic_loader.cc]: MKLDNN does not support?

File: paddle/fluid/platform/dynload/tensorrt.h [hunk 8]
```diff
@@ -32,21 +32,24 @@ extern void* tensorrt_dso_handle;
...
+  struct DynLoad__##__name {                                               \
+    template <typename... Args>                                            \
+    auto operator()(Args... args) -> DECLARE_TYPE(__name, args...) {       \
+      using tensorrt_func = decltype(&::__name);                           \
+      std::call_once(tensorrt_dso_flag, []() {                             \
+        tensorrt_dso_handle =                                              \
+            paddle::platform::dynload::GetTensorRtDsoHandle();             \
+        PADDLE_ENFORCE_NOT_NULL(                                           \
+            tensorrt_dso_handle,                                           \
+            platform::errors::Unavaliable("load tensorrt so failed"));     \
```
Comment (chenwhql) [inline paddle/fluid/platform/dynload/tensorrt.h]: may be here using the so file name is better

File: paddle/fluid/platform/cpu_helper.cc (hunk 1 above)
Comment (GaoWei8) [reply paddle/fluid/platform/cpu_helper.cc]: Done.

File: paddle/fluid/platform/cudnn_helper.h (hunk 2 above)
Comment (GaoWei8) [reply paddle/fluid/platform/cudnn_helper.h]: Done.

File: paddle/fluid/platform/cudnn_helper.h (hunk 3 above)
Comment (GaoWei8) [reply paddle/fluid/platform/cudnn_helper.h]: Done.

File: paddle/fluid/platform/cudnn_helper.h (hunk 4 above)
Comment (GaoWei8) [reply paddle/fluid/platform/cudnn_helper.h]: Done.

File: paddle/fluid/platform/cudnn_helper.h (hunk 5 above)
Comment (GaoWei8) [reply paddle/fluid/platform/cudnn_helper.h]: Done.

File: paddle/fluid/platform/device_context.cc (hunk 6 above)
Comment (GaoWei8) [reply paddle/fluid/platform/device_context.cc]: Done.

File: paddle/fluid/platform/dynload/dynamic_loader.cc (hunk 7 above)
Comment (GaoWei8) [reply paddle/fluid/platform/dynload/dynamic_loader.cc]: Done.

File: paddle/fluid/platform/dynload/tensorrt.h (hunk 8 above)
Comment (GaoWei8) [reply paddle/fluid/platform/dynload/tensorrt.h]: Define函数内部的PADDLE_ENFORCE统一在下一个PR中修改。

**Pull request merged**